kroxylicious/src/main/java/io/kroxylicious/proxy/KafkaProxy.java

```diff
@@ -88,28 +90,44 @@ public KafkaProxy startup() throws InterruptedException {
             throw new IllegalStateException("This proxy is already running");
         }
 
+        var portConflictDefector = new PortConflictDetector();
+        portConflictDefector.validate(virtualClusterMap);
+
         var availableCores = Runtime.getRuntime().availableProcessors();
         var meterRegistries = new MeterRegistries(micrometerConfig);
 
-        this.adminEventGroup = buildNettyEventGroups(availableCores, config.isUseIoUring());
-        this.serverEventGroup = buildNettyEventGroups(availableCores, config.isUseIoUring());
+        this.adminEventGroup = buildNettyEventGroups("admin", availableCores, config.isUseIoUring());
+        this.serverEventGroup = buildNettyEventGroups("server", availableCores, config.isUseIoUring());
 
         maybeStartMetricsListener(adminEventGroup, meterRegistries);
 
-        this.endpointProviders = virtualClusterMap.entrySet().stream()
-                .collect(Collectors.toMap(e -> e.getValue().getClusterEndpointProvider(), Map.Entry::getValue));
-
-        var networkBindings = buildNetworkBindings();
+        var tlsServerBootstrap = buildServerBootstrap(serverEventGroup, new KafkaProxyInitializer(config, true, endpointRegistry, false, Map.of()));
+        var plainServerBootstrap = buildServerBootstrap(serverEventGroup, new KafkaProxyInitializer(config, false, endpointRegistry, false, Map.of()));
 
-        var tlsServerBootstrap = buildServerBootstrap(serverEventGroup, new KafkaProxyInitializer(config, true, this, false, Map.of()));
-        var plainServerBootstrap = buildServerBootstrap(serverEventGroup, new KafkaProxyInitializer(config, false, this, false, Map.of()));
+        var unused = networkBindingExecutor.submit(() -> {
+            try {
+                do {
+                    var networkBindingOperation = endpointRegistry.takeNetworkBindingEvent();
+                    var bootstrap = networkBindingOperation.tls() ? tlsServerBootstrap : plainServerBootstrap;
+                    networkBindingOperation.performBindingOperation(bootstrap, networkBindingExecutor);
+                } while (!(Thread.interrupted() || endpointRegistry.isRegistryClosed()));
+            }
```

Comment:
Does this loop ever complete in reality as `endpointRegistry.takeNetworkBindingEvent()` blocks indefinitely so there are rarely opportunities for this test to be evaluated.